### PR TITLE
#3178771 drupal 9 compatibility

### PR DESCRIPTION
**Pull request by krystalcode**
Drupal.org issue: https://www.drupal.org/project/commerce_price_rule/issues/3178771

To use this branch add the following to the `repositories` section in your `composer.json` file:
```
"repositories": {
     "commerce-price-rule": {                                                                                                                                                                                                         
           "type": "vcs",                                                                                                                                                                                                               
            "url": "https://github.com/krystalcode/drupal8-commerce-price-rule"                                                                                                                                                          
     },
}
```
And then:
```
"require": {
    "drupal/commerce_price_rule": "dev-3178771-drupal-9-compatibility
}
```

### CODE REVIEW

File: commerce_price_rule.info.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Commerce Price Rule
 type: module
 description: 'Provides functionality for personalising product prices based on chosen conditions.'
 package: Commerce (contrib)
-core: 8.x
+core_version_requirement: ^8.7.7 || ^9
 dependencies:
   - commerce:commerce_product
-  - system (>=8.3.0)
+  - commerce:commerce_order
```
Comment (krystalcode) [inline commerce_price_rule.info.yml]: **CHANGE REQUEST**

@barinder Were you getting a particular error? As far as I see the module does not depend on orders.

Comment (barinder) [reply commerce_price_rule.info.yml]: @krystalcode Yes, if commerce_order is not enabled following fatal error is thrown while adding price rule.
`"Drupal\Component\Plugin\Exception\PluginNotFoundException: The "commerce_order_item" entity type does not exist. in Drupal\Core\Entity\EntityTypeManager->getDefinition() (line 150 of /Users/barinder/Sites/d9/core/lib/Drupal/Core/Entity/EntityTypeManager.php)."`

Comment (krystalcode) [reply commerce_price_rule.info.yml]: @barinder Ok I just got some time to look into this. This is a bug in the Commerce module; I have created an [issue](https://www.drupal.org/project/commerce/issues/3184470) and it should be fixed there. We should not be demanding the order module here.

**Comment by krystalcode** [comment]
Merged in Drupal.org.

**Pull request closed without merging**